jena-text/src/main/java/org/apache/jena/query/text/assembler/TextIndexLuceneAssembler.java

```diff
@@ -94,7 +94,17 @@ public TextIndex open(Assembler a, Resource root, Mode mode) {
                 }
                 isMultilingualSupport = mlsNode.asLiteral().getBoolean();
             }
-            
+
+            int maxBasicQueries = 1024;
+            Statement maxBasicQueriesStatement = root.getProperty(pMaxBasicQueries);
+            if (null != maxBasicQueriesStatement) {
+                RDFNode mbqNode = maxBasicQueriesStatement.getObject();
+                if (! mbqNode.isLiteral()) {
+                    throw new TextIndexException("text:maxBasicQueries property must be a int : " + mbqNode);
```

Comment:
All the code looks good to me. Only one question here. The message says that it must be an int. But up to this point, I think we only know it's a) not null, and b) not a literal.

My RDF-Fu is a bit rusty, feel free to ignore if what I'm saying is completely wrong please. But I think it could be another type (char, bool, etc)? So I think only after this `if`, when we call `getInt()` we will actually know whether we have an `int` or not?

Reply:
Good point - this gives a basic warning for the first method call in `maxBasicQueries = mbqNode.asLiteral().getInt()`

`getInt` also parses xsd:String as a 32-bit integer (all the `get` do similar)

Having both the `isLiteral` test and the try-catch means better error messages.

The whole block could be wrapped in a try-catch for robustness. I'll add it in the post-merge work.